config/src/test/java/org/hyperledger/besu/config/JsonGenesisConfigOptionsTest.java

```diff
@@ -0,0 +1,103 @@
+      final String configText =
+          Resources.toString(
+              Resources.getResource("valid_config_with_custom_forks.json"), StandardCharsets.UTF_8);
+      return JsonUtil.objectNodeFromString(configText);
+    } catch (final IOException e) {
+      throw new RuntimeException("Failed to load resource", e);
+    }
+  }
+
+  @Test
+  public void customForksDecodesCorrectlyFromFile() {
+    final ObjectNode configNode = loadCompleteDataSet();
+
+    final JsonGenesisConfigOptions configOptions =
+        JsonGenesisConfigOptions.fromJsonObject(configNode);
+
+    assertThat(configOptions.getCustomForks()).isNotNull();
+    assertThat(configOptions.getCustomForks().getIbftForks().size()).isEqualTo(2);
+    assertThat(configOptions.getCustomForks().getIbftForks().get(0).getForkBlock()).isEqualTo(20);
+    assertThat(configOptions.getCustomForks().getIbftForks().get(0).getValidators()).isNotEmpty();
+    assertThat(configOptions.getCustomForks().getIbftForks().get(0).getValidators().get())
+        .containsExactly(
+            "0x12345678901234567890123456789012345678900x1234567890123456789012345678901234567890",
+            "0x98765432109876543210987654321098765432100x9876543210987654321098765432109876543210");
+
+    assertThat(configOptions.getCustomForks().getIbftForks().get(1).getForkBlock()).isEqualTo(25);
+    assertThat(configOptions.getCustomForks().getIbftForks().get(1).getValidators()).isNotEmpty();
+    assertThat(configOptions.getCustomForks().getIbftForks().get(1).getValidators().get())
+        .containsExactly(
+            "0x12345678901234567890123456789012345678900x1234567890123456789012345678901234567890");
+  }
+
+  @Test
+  public void configWithMissingCustomForksIsValid() {
+    final ObjectNode configNode = loadCompleteDataSet();
+    configNode.remove("customForks");
+
+    final JsonGenesisConfigOptions configOptions =
+        JsonGenesisConfigOptions.fromJsonObject(configNode);
+
+    assertThat(configOptions.getCustomForks()).isNotNull();
+    assertThat(configOptions.getCustomForks().getIbftForks().size()).isZero();
+  }
+
+  @Test
+  public void configWithEmptyCustomForksIsValid() {
+    final ObjectNode configNode = loadCompleteDataSet();
+    final ObjectNode customForksNode = JsonUtil.getObjectNode(configNode, "customForks").get();
+    customForksNode.remove("ibft2");
+
+    final JsonGenesisConfigOptions configOptions =
+        JsonGenesisConfigOptions.fromJsonObject(configNode);
+
+    assertThat(configOptions.getCustomForks()).isNotNull();
+    assertThat(configOptions.getCustomForks().getIbftForks().size()).isZero();
+  }
+
+  @Test
+  public void configWithAnIbftWithNoValidatorsListedIsValid() {
+    final ObjectNode configNode = loadCompleteDataSet();
+    final ObjectNode customForksNode = JsonUtil.getObjectNode(configNode, "customForks").get();
+    final ArrayNode ibftNode = JsonUtil.getArrayNode(customForksNode, "ibft2").get();
+    ((ObjectNode) ibftNode.get(0)).remove("validators");
```

Comment:
nit: encapsulating this logic into a private method describing what is being done would improve readability.

e.g. `final ObjectNode configNode = loadConfigWithNoValidatorsListed()`

The same applies to some of the other tests in this class.

Reply:
Done.